pkg/auth/iam/iam.go

```diff
@@ -320,27 +335,30 @@ func WithValidScope(scope string) FilterOption {
 }
 
 // WithMatchedSubdomain filters request to a subdomain to match it with namespace in user's token
+// Deprecated: subdomain validation will be done in the auth filter
 func WithMatchedSubdomain(excludedNamespaces []string) FilterOption {
 	return func(req *restful.Request, iamClient iam.Client, claims *iam.JWTClaims) error {
-		part := strings.Split(getHost(req.Request), ".")
-		if len(part) < 3 {
-			// url with subdomain should have at least 3 part, e.g. foo.example.com, otherwise we should not check it
-			return nil
-		}
-
-		for _, excludedNS := range excludedNamespaces {
-			if strings.ToLower(excludedNS) == strings.ToLower(claims.Namespace) {
-				return nil
-			}
-		}
+		return matchSubdomainWithNamespace(getHost(req.Request), claims.Namespace, excludedNamespaces)
+	}
+}
 
-		if strings.ToLower(claims.Namespace) == strings.ToLower(part[0]) {
+func matchSubdomainWithNamespace(host string, namespace string, excludedNamespaces []string) error {
```

Comment:
could we refactor this to return boolean instead of error,

so from the caller side, we could simply make it like this
```
if valid := matchSubdomainWithNamespace(getHost(req.Request), claims.Namespace, filter.options.SubdomainValidationExcludedNamespaces); !valid {
  logIfErr(resp.WriteHeaderAndJson(http.StatusNotFound, ErrorResponse{
    ErrorCode:    SubdomainMismatch,
    ErrorMessage: ErrorCodeMapping[SubdomainMismatch],
  }, restful.MIME_JSON))
}
```

Reply:
@marselsampe I have updated the PR, please help review again